### PR TITLE
Fixed dynamic non generated routes, updated documentation, cleaner build config.

### PR DESCRIPTION
**Pull request by Livog**
Added generate folder in nuxt.config.js insead of package.json
Fixed dynamic routes for non generated routes.
Updated documentation for helping non zeit/now projects to convert their projects easier.

### CODE REVIEW

File: nuxtjs/now.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "routes": [
```
Comment (paulogdm) [inline nuxtjs/now.json]: Why we don't use the default behavior here? https://zeit.co/docs/v2/advanced/routes/

File: nuxtjs/nuxt.config.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 export default {
-  mode: 'spa',
+  mode: 'universal',
```
Comment (paulogdm) [inline nuxtjs/nuxt.config.js]: Why did you change it from SPA to universal?

Comment (Livog) [reply nuxtjs/nuxt.config.js]: **SPA:** We leave it up to the browser to render the content. Each path / index.html file just tells the browser what javascript it needs to render the page. The content of the page is embedded into the javascript files, not in the Html files themselves.
**Universal:** This is the beauty of Nuxt, the content is embedded into the Html files. These files also contain var window.__NUXT__ with all the data sent from the server. Nuxt uses this to hydrate the content with Vue and all the components and makes the page reactive. This is much better for SEO since Search Engines are not the best at reading Javascript generated content.

But of course, I can't know the goal and purpose of Zeit Now, maybe SPA is the only thing that's supposed to be available to host.

File: nuxtjs/now.json (hunk 1 above)
Comment (Livog) [reply nuxtjs/now.json]: When you build apps/ websites with nuxt. It's very common to rely on an outside source for content. Such as using Wordpress REST API to retrieve content, or Firebase, etc. So if we would make a path that relays on order data and to get it by ID. e.g `/order/${$route.params.id}`

This path can be created by adding /pages/order/_id.vue

We can, of course, choose to generate these routes in nuxt.config.js
```
generate: {
    routes: [
      '/order/1',
      '/order/2',
      '/order/3'
    ]
  }
```

Sometimes we don't even want to generate order at all. We would like to do some verification before we serve the data.

But if you would not have this now.json file and you go to a path that we have NOT generated. Since it's dynamic you will get an error. If you navigate to this page it's fine since Nuxt is doing the loading and swapping components. This is so it falls back on the SPA when the content is not generated. Then let the SPA serve a 404 if we don't find the data. 
Maybe there is a better way of configuring this up that I don't know of.

**Comment by Livog** [comment]
@paulogdm Is see that you took the code that I wrote and implemented it in another push request. https://github.com/zeit/now-examples/pull/463. I have made valid points in the pull request. Let me know what you think.

**Comment by msweeneydev** [comment]
Hello @Livog, thanks for your suggestions here, they are much appreciated!

As you mentioned, I've amended the output directory in `nuxt.config.js`, this clearly makes much more sense than moving the build output.

As for using `universal`, this is something we will look to discuss soon, I need to do some more testing with it first of all. Including a default route is something to consider, however we may look at including this as an optimization for Nuxt.js - another think we'll discuss shortly!

I'll be looking to get back to you next week on this so we can find the best path forward for Nuxt.js on Now.

File: nuxtjs/nuxt.config.js (hunk 2 above)
Comment (msweeneydev) [reply nuxtjs/nuxt.config.js]: @Livog at present, only `nuxt generate` (SPA) is available. When Nuxt.js supports auto-export, Universal should be possible.

**Comment by msweeneydev** [comment]
Hello @Livog, there have been a few changes made to this example recently, it will now reside in the [zeit/now](https://github.com/zeit/now/tree/master/examples) repository.

Thanks again for making this PR, I'll make sure these changes, if still applicable, get carried over.
